variant/tokenizers/amino_acid_substitution.py

```diff
@@ -0,0 +1,78 @@
+"""A module for Amino Acid Substitution Tokenization."""
+from typing import Optional
+from variant.schemas.token_response_schema import AminoAcidSubstitutionToken,\
+    TokenMatchType
+from .polypeptide_sequence_variant_base import PolypeptideSequenceVariantBase
+
+
+class AminoAcidSubstitution(PolypeptideSequenceVariantBase):
+    """Class for tokenizing Amino Acid Substitution."""
+
+    def match(self, input_string) -> Optional[AminoAcidSubstitutionToken]:
+        """Return a AminoAcidSubstitutionToken match if one exists.
+
+        :param str input_string: The input string to match
+        :return: A AminoAcidSubstitutionToken if a match exists.
+            Otherwise, None.
+        """
+        input_string = str(input_string)
+
+        psub_parts = None
+        self.psub = {
+            'amino_acid': None,
+            'position': None,
+            'new_amino_acid': None
+        }
+
+        if input_string is None:
+            return None
+
+        if '.' in input_string:
+            if not input_string.startswith('p.'):
+                return None
+            p_count = input_string.count('p.')
+            if p_count == 1:
+                psub_parts = self.splitter.split(input_string)
+            elif p_count == 2:
+                psub_parts = input_string.split()
+        else:
+            psub_parts = self.splitter.split(input_string)
+
+        self._get_psub(psub_parts)
+
+        if all(self.psub.values()):
```

Comment:
Is there something specific that this is checking (ie like a blank string?)

Also idk if this is a realistic example, but I believe that if `psub['position']` is 0, this will evaluate to False (since `bool(0)` is False).

Reply:
Ope I remember running into this problem early on and didn't make a note to fully fix. Thanks for reminding me!